streaming/base/dataset.py

```diff
@@ -103,6 +104,22 @@ def __init__(self, sample_ids: NDArray[np.int64]) -> None:
         self._state = 0
         self._num_exited = 0
 
+        if sys.version_info[1] <= 8:
```

Comment:
Can you please add a comment here why we had to made this changes ? This will make a code more readable. Thank You!

Reply:
sure, added.